test/firestore_test.dart

```diff
@@ -63,6 +63,30 @@ void main() {
         expect(nested.getString('author'), 'Isaac Asimov');
       });
 
+      test('get set', () {
```

Comment:
This test seems to be duplicating logic from other test cases. Though I understand this is a more lightweight unit test it still feels a bit redundant.

For the `FieldValue` edge cases I'd create separate test cases for each value as integration tests, writing to the database and reading back the results.

Reply:
Oh, I see you actually have separate tests for special field values below. I'm a bit confused about the purpose of this test than. :)

Reply:
Many setters (setDouble, setBool) did not have any unit test. The goal is to make sure that we can read/get all fields with setter and getter and that the roundTrip fromMap(toMap) was working fine too